<launch>
  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
	<include file="$(find gazebo_ros)/launch/empty_world.launch">
		<arg name="world_name" value="$(find hku_m100_gazebo)/worlds/sphere_world2.world"/>
		<!-- more default parameters can be changed here -->
		<arg name="verbose" value="true"/>
		<arg name="paused" value="false"/>
		<arg name="use_sim_time" value="true"/>
		<arg name="gui" value="true"/>
		<arg name="recording" value="false"/>
		<arg name="debug" value="false"/>
	</include>

	<rosparam param="nr_of_obstacles">100</rosparam>
	

		<!--<node pkg="hku_m100_gazebo" type="move_spheres" name="move_spheres" output="screen"/> -->
	<!-- Start Bridge Node with roslaunch rather than bash -->
	<node name="hku_m100_pcsim_gazebo_bridge" pkg="hku_m100_gazebo" type="hku_m100_pcsim_gazebo_bridge" respawn="true" /> 
	<!-- Start rviz visualization with preset config -->
	 <!--<arg name="model" default="$(find hku_m100_description)/urdf/hku_m100_base_with_gimbal.urdf.xacro"/>
	<arg name="name" default="hku_m100"/>
	<arg name="world_frame" default="/world"/>
	<param name="robot_description"
          command="$(find xacro)/xacro - -inorder '$(arg model)'
                      hitl_model:=false"/>
	<arg name="marker_size" default="27.2" />
	<arg name="max_new_marker_error" default="0.08" />
	<arg name="max_track_error" default="0.2" />
	<arg name="cam_image_topic" default="/front_cam/camera/image" />
	<arg name="cam_info_topic" default="/front_cam/camera/camera_info" />	
	<arg name="output_frame" default="/front_cam_link" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find hku_m100_gazebo)/rviz_cfg/flight.rviz"/> -->

<!--	<node pkg="hku_m100_gazebo" type="rviz_collision_publisher" name="rviz_collision_publisher" output="screen"/> -->

  
  <!-- Start rviz visualization with preset config -->
  <!--<node pkg="rviz" type="rviz" name="rviz" args="-d $(find hector_quadrotor_demo)/rviz_cfg/outdoor_flight.rviz"/>-->
<!--  	<arg name="name" default="hku_m100"/>
	<arg name="world_frame" default="world"/>
	<arg name="cam_image_topic" default="camera/image" />
	<arg name="cam_info_topic" default="camera/camera_info" />	
	<arg name="output_frame" default="/camera_optical_frame" />
	<arg name="tf_prefix" default="hku_m100" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find hku_m100_gazebo)/rviz_cfg/flight.rviz"/> -->
	
    <!-- <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0.0 0.0 0.0 0.0 0.0 0.0 0.0 /world /world_frame 100" /> -->
    
	<!-- <node pkg="hku_m100_gazebo" type="psu_m100_gasSensor" name="psu_m100_gasSensor" output="screen"/> -->


	

</launch>
